File: examples/cpp/01_cpp_minimal/is_prime_test.cpp
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

#include "is_prime.hpp"

#include "touca/touca.hpp"

int main(int argc, char* argv[]) {
  touca::workflow("is_prime", [](const std::string& testcase) {
    const auto number = std::stoul(testcase);
    touca::check("output", is_prime(number));
  });
  touca::run(argc, argv);
}

File: examples/js/01_node_minimal/is_prime_test.ts
// Copyright 2022 Touca, Inc. Subject to Apache-2.0 License.

import { touca } from '@touca/node';
import { is_prime } from './is_prime.js';

touca.workflow('is_prime_test', (testcase: string) => {
  const number = Number.parseInt(testcase);
  touca.check('is_prime_output', is_prime(number));
});

touca.run();
